profitbricks-rest/src/main/java/org/apache/jclouds/profitbricks/rest/compute/ProfitBricksComputeServiceAdapter.java

```diff
@@ -0,0 +1,577 @@
+               input = api.lanApi().get(dataCenterId, input.id(), new DepthOptions().depth(3));
+               return input.properties().isPublic();
+            }
+         }).orNull();
+      }
+      if (lan == null) {
+         logger.warn("Could not find an existing lan Creating one....");
+         lan = api.lanApi().create(Lan.Request.creatingBuilder()
+                 .dataCenterId(dataCenterId)
+                 .isPublic(Boolean.TRUE)
+                 .name("lan " + name)
+                 .build());
+      }
+
+      //add a NIC to the server
+      int lanId = DEFAULT_LAN_ID;
+      if (options.getNetworks() != null) {
+         try {
+            String networkId = Iterables.get(options.getNetworks(), 0);
+            lanId = Integer.valueOf(networkId);
+         } catch (Exception ex) {
+            logger.warn("no valid network id found from options. using default id='%d'", DEFAULT_LAN_ID);
+         }
+      }
+
+      Nic nic = api.nicApi().create(Nic.Request.creatingBuilder()
+              .dataCenterId(dataCenterId)
+              .name("jclouds" + name)
+              .dhcp(Boolean.TRUE)
+              .lan(lanId)
+              .firewallActive(Boolean.FALSE)
+              .serverId(serverId).
+              build());
+
+      waitNICUntilAvailable.apply(NicRef.create(dataCenterId, serverId, nic.id()));
+      waitDcUntilAvailable.apply(dataCenterId);
+      waitServerUntilAvailable.apply(ServerRef.create(dataCenterId, serverId));
+
+      //connect the rest of volumes to server;delete if fails
+      final int volumeCount = volumeIds.size();
+      for (int j = 1; j < volumeCount; j++) { // skip first; already connected
+         final String volumeId = volumeIds.get(j);
+         try {
+            logger.trace("<< connecting volume '%s' to server '%s'", volumeId, serverId);
+            provisioningManager.provision(jobFactory.create(group, new Supplier<Object>() {
+
+               @Override
+               public Object get() {
+                  return api.serverApi().attachVolume(
+                          Server.Request.attachVolumeBuilder()
+                          .dataCenterId(dataCenterId)
+                          .serverId(serverId)
+                          .volumeId(volumeId)
+                          .build()
+                  );
+               }
+            }));
+
+            logger.trace(">> volume connected.");
+         } catch (Exception ex) {
+            try {
+               // delete unconnected volume
+               logger.warn(ex, ">> failed to connect volume '%s'. deleting..", volumeId);
+               destroyVolume(volumeId, dataCenterId);
+               logger.warn(ex, ">> rolling back server..", serverId);
+               destroyServer(serverId, dataCenterId);
+               throw ex;
+            } catch (Exception ex1) {
+               java.util.logging.Logger.getLogger(ProfitBricksComputeServiceAdapter.class.getName()).log(Level.SEVERE, null, ex1);
```

Comment:
Updated to the latest just let me know if you need me to squash this, Thanks for the review.